joomla/joomla-cms
# TinyMCE improvements

Pull Request for Issue # .

### Summary of Changes
- This PR is patching a bug in tooltips misplacement after using drag and drop
- It moves the drag and drop plugin out of the plugins directory (easier maintainability)
- It removes the page break plugin that comes with tinyMCE and introduces a custom plugin that creates a Joomla compatible `<hr id="readmore"/>`
- It also deprecates the option for selecting a skin and introduces an automatic skin selector using the current template. The theme should be bound to the template :)

### Testing Instructions
Before applying this patch make sure that tinyMCE is your current editor and the drag and drop is enabled. Also make sure that page break button is placed in your toolbar.

Observe the bugs:
- Drag and drop an image, after the image has been uploaded try hovering in any button in the toolbar, the tooltip is way off.
- Click on the page break and the readmore buttons, check the produced code

Apply the patch

Repeat the above steps and verify that tooltips are displaying correctly and page break produces the same code as read more!

Download a skin for tinymce e.g. https://github.com/dgt41/bootstrap-test/raw/gh-pages/editors.zip_.zip
Extract the skin and place it in:
`administrator/templates/isis/css`
and
`templates/protostar/css`
Visit the article form in back end and front end. Verify that the skin is loaded

### Documentation Changes Required
The skin part needs to be documented somewhere in the templates docs.

For J4 the automation script for asset management needs to delete the page break plugin!
